core/connection/manager.go

```diff
@@ -149,7 +145,25 @@ func (manager *connectionManager) startConnection(consumerID identity.Identity,
 	}
 	cancel = append(cancel, func() { dialog.Close() })
 
-	sessionID, sessionConfig, err := session.RequestSessionCreate(dialog, proposal.ID)
+	stateChannel := make(chan State, 10)
+	statisticsChannel := make(chan consumer.SessionStatistics, 10)
+
+	connection, err := manager.newConnection(proposal.ServiceType, stateChannel, statisticsChannel)
+	if err != nil {
+		return err
+	}
+
+	sessionCreateConfig, err := connection.GetSessionConfig()
+	if err != nil {
+		return err
+	}
+
+	sessionCreateConfigJSON, err := json.Marshal(sessionCreateConfig)
```

Comment:
I would leave session marshaling to underlying mechanism, and pass returned consumer config structure directly